src/Kernel32.Shared/Win32Exception.cs

```diff
@@ -46,7 +46,7 @@ public Win32Exception()
         /// <param name="error">The Win32 error code associated with this exception.</param>
         public Win32Exception(Win32ErrorCode error)
 #if DESKTOP
-            : base((int)error)
+            : this(error, error.GetMessage() ?? $"Unknown error (0x{(int)error:x8})")
```

Comment:
When I say "the OS returns the current language" I mean that we are calling `status.GetMessage()` which calls Win32 API `FormatMessage` from the OS and not the .NET Framework, thus the call returns by default (if you don't specify) the default UI language. It does not seem to be a .NET behavior because we are not using it for this end.

It might be that another .NET behavior plays a trick on us on this. I can't run the tests inside VS to check the behavior because I have lots of errors, I have never been able to compile successfully insisde VS just using the command prompt script. It could be that under the command prompt console .NET uses the UI language for the thread and ignores when we change (I know the console threads have a different behavior than regular .NET threads).

Anyway, we are not using .NET methods to get the message that's why setting the thread language inside .NET does not work on these tests.